lib/compute/snapshot.js

```diff
@@ -50,13 +59,93 @@ var util = require('../common/util.js');
  * var gce = gcloud.compute();
  *
  * var snapshot = gce.snapshot('snapshot-name');
+ *
+ * //-
+ * // Or, access through a disk.
+ * //-
+ * var disk = gce.zone('us-central1-a').disk('disk-name');
+ * var snapshot = disk.snapshot('disk-snapshot-name');
  */
-function Snapshot(compute, name) {
-  this.compute = compute;
-  this.name = name;
-  this.metadata = {};
+function Snapshot(scope, name) {
+  var isDisk = scope.constructor.name === 'Disk';
```

Comment:
@callmehiphop here's a place -- a Snapshot can be instantiated from a Disk (`compute().disk().snapshot()`) or the core `compute().snapshot()`. A Snapshot must be paired with a Disk in order to be created. Similar to how an Instance (VM) can't be created without a Zone. To enable these methods only where appropriate, we sniff if this scope is a Disk.